packages/gatsby/src/bootstrap/load-plugins/validate.ts

```diff
@@ -167,77 +172,76 @@ export async function handleBadExports({
   }
 }
 
-export async function validatePluginOptions({
-  flattenedPlugins,
-}: {
-  flattenedPlugins: Array<IPluginInfo & Partial<IFlattenedPlugin>>
-}): Promise<void> {
-  const errors = (
-    await Promise.all(
-      flattenedPlugins.map(
-        async (plugin): Promise<boolean | null> => {
-          if (plugin.nodeAPIs?.indexOf(`pluginOptionsSchema`) === -1)
-            return null
-
-          const gatsbyNode = require(`${plugin.resolve}/gatsby-node`)
-          if (!gatsbyNode.pluginOptionsSchema) return null
-
-          let optionsSchema = (gatsbyNode.pluginOptionsSchema as Exclude<
-            GatsbyNode["pluginOptionsSchema"],
-            undefined
-          >)({
-            Joi,
+export async function validatePluginOptions(
+  config: ISiteConfig = {}
+): Promise<void> {
+  if (!config.plugins) return
+
+  let errors = 0
+
+  config.plugins = await Promise.all(
+    config.plugins.map(async plugin => {
+      let gatsbyNode
+
+      try {
+        gatsbyNode = require(`${plugin.resolve}/gatsby-node`)
+      } catch (err) {
+        gatsbyNode = {}
+      }
+
+      if (!gatsbyNode.pluginOptionsSchema) return plugin
+
+      let optionsSchema = (gatsbyNode.pluginOptionsSchema as Exclude<
+        GatsbyNode["pluginOptionsSchema"],
+        undefined
+      >)({
+        Joi,
+      })
+
+      // Validate correct usage of pluginOptionsSchema
+      if (!Joi.isSchema(optionsSchema) || optionsSchema.type !== `object`) {
+        reporter.warn(
+          `Plugin "${plugin.resolve}" has an invalid options schema so we cannot verify your configuration for it.`
+        )
+        return plugin
+      }
+
+      try {
+        if (!optionsSchema.describe().keys.plugins) {
+          // All plugins have "plugins: []"" added to their options in load.ts, even if they
+          // do not have subplugins. We add plugins to the schema if it does not exist already
+          // to make sure they pass validation.
+          optionsSchema = optionsSchema.append({
+            plugins: Joi.array().length(0),
+          })
+        }
+
+        plugin.options = await validateOptionsSchema(
+          optionsSchema,
+          (plugin.options as IPluginInfoOptions) || {}
+        )
```

Comment:
We're missing sub plugin validation for things like gatsby-transform-remark
```suggestion
        plugin.options = await validateOptionsSchema(
          optionsSchema,
          (plugin.options as IPluginInfoOptions) || {}
        )

        if (plugin.options.plugins) {
          await validatePluginOptions({ plugins: plugin.options.plugins })
        }
```